### DISCUSSION TITLE
Why my map flashing or sometimes missing shadow in editing mode?

### DISCUSSION THREAD

**Discussion by ghost** [redeclipse/base#1240] [Help and Support]
I am not set to do this.

[maps.zip](https://github.com/redeclipse/base/files/6035569/maps.zip)

**Comment by cloudziq** [comment]
the main reason for this:  Too many visible lights in view.
you using way too many light entities on this kind of map where occlusion is not working very well
I checked the map and I found that every light texture have one or even a few light ents
How to fix?:   
  -Use fewer lights at all ;p
  -put one light entity instead of a few, close ones (1 light entity per few light fixtures)
  
You can look how lighting is done on other maps which are included in game :)

**Comment by SniperGoth** [comment]
What happened, is that the shadow atlas got way too busy and doesn't know what to do or what to cache, so it keeps flickering because of that.
One way to fix this, is to set lights that aren't necessary to have shadows, to no-shadow in the flags attribute.
Or as dziq said, have one big light instead of a bunch of tiny ones, depending on the case.
